openquake/hazardlib/gsim/base.py

```diff
@@ -286,20 +290,29 @@ def make_contexts(self, site_collection, rupture):
             and distance parameters) is unknown.
         """
         rctx = self.make_rupture_context(rupture)
-        distances = get_distances(rupture, site_collection.mesh, 'rjb')
-        sites = site_collection
-        if self.maximum_distance:
-            mask = distances <= self.maximum_distance
-            if mask.any():
-                sites = site_collection.filter(mask)
-                distances = distances[mask]
-            else:
-                raise FarAwayRupture
-
+        sites, distances = self.get_closest(site_collection, rupture, 'rjb')
         sctx = self.make_sites_context(sites)
         dctx = self.make_distances_context(sites, rupture, {'rjb': distances})
         return (sctx, rctx, dctx)
 
+    def get_closest(self, sites, rupture, distance_type='rjb'):
+        """
+        :param rupture: a rupture
```

Comment:
Please swap rupture and sites, to have them in the same order as in the signature